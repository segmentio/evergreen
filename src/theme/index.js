export { default as defaultTheme } from './src/default-theme'
export {
  default as ThemeContext,
  ThemeProvider,
  ThemeConsumer
} from './src/ThemeContext'
export { default as withTheme } from './src/withTheme'
export { default as useTheme } from './src/useTheme'
