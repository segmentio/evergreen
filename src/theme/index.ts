export { default as ThemeContext, getThemeContext, ThemeProvider, ThemeConsumer } from './src/ThemeContext'
export { default as withTheme } from './src/withTheme'
export { default as useTheme } from './src/useTheme'
export { get, getValue, mergeTheme, resolveThemeTokens } from './src/theme-tools'
