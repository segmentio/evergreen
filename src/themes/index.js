export { default as defaultTheme } from './default'
