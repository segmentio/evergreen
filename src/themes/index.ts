export { default as defaultTheme } from './default'
export type {
  Components,
  DefaultTheme,
  DefaultThemeAppearances,
  DefaultThemeColors,
  DefaultThemeColor,
  DefaultThemeFills,
  DefaultThemeIntents,
  DefaultThemeIntent,
  DefaultThemePseudoSelectors,
  DefaultThemeSizes,
} from './default'
export { default as classicTheme } from './classic'
export { default as deprecatedDefaultTheme } from './deprecated'
