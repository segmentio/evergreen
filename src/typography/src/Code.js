import React, { PureComponent } from 'react'
import Text from './Text'

export default class Code extends PureComponent {
  static propTypes = {
    ...Text.propTypes
  }

  render() {
    return <Text is="code" fontFamily="mono" {...this.props} />
  }
}
