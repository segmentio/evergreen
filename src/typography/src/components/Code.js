import React, { PureComponent } from 'react'
import Text from './Text'

export default class Code extends PureComponent {
  static propTypes = {
    ...Text.propTypes
  }

  static defaultProps = {
    ...Text.defaultProps,
    is: 'code',
    fontFamily: 'mono'
  }

  render() {
    return <Text {...this.props} />
  }
}
