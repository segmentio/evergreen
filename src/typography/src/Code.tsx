import React, { PureComponent } from 'react'
import cx from 'classnames'
import { withTheme, Theme } from '../../theme'
import Text from './Text'

interface CodeProps extends React.ComponentProps<typeof Text> {
  /**
   * The appearance of the code.
   */
  appearance: 'default' | 'minimal'

  /**
   * Class name passed to the button.
   * Only use if you know what you are doing.
   */
  className?: string

  /**
   * Theme provided by ThemeProvider.
   */
  theme: Theme
}

class Code extends PureComponent<CodeProps> {
  static defaultProps: Partial<CodeProps> = {
    appearance: 'default'
  }

  render() {
    const { theme, className, appearance, ...props } = this.props

    const {
      className: themedClassName = '',
      ...themeProps
    } = theme.getCodeProps(appearance)

    return (
      <Text
        is="code"
        className={cx(className, themedClassName)}
        fontFamily="mono"
        {...themeProps}
        {...props}
      />
    )
  }
}

export default withTheme(Code)
