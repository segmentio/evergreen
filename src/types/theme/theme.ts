import { Components } from '../../themes'
import { Color } from './color'
import { ComponentStyles } from './component-styles'
import { Fill } from './fill'
import { FontFamilies } from './font-families'
import { FontWeights } from './font-weights'
import { Intent } from './intent'
import { LettingSpacings } from './letter-spacings'
import { ZIndices } from './z-indices'

export interface Theme<TComponents extends Components = Components> {
  colors: { [color: string]: Color<string | { [group: string]: Color }> }
  fills: { [fill: string]: Fill }
  intents: { [intent: string]: Intent }
  fontFamilies: FontFamilies
  radii: string[]
  shadows: string[] & { focusRing: string }
  fontSizes: string[]
  fontWeights: FontWeights
  letterSpacings: LettingSpacings
  lineHeights: string[]
  zIndices: ZIndices
  components: Partial<ComponentStyles<TComponents>>
}
