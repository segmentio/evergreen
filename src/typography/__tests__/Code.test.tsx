import React from 'react'
import { render } from '@testing-library/react'
import faker from 'faker'
import renderer from 'react-test-renderer'
import { UIBoxSerializer } from '../../../lib/testing'
import { ThemeProvider } from '../../theme'
import { defaultTheme } from '../../themes'
import Code from '../src/Code'

expect.addSnapshotSerializer(UIBoxSerializer)

describe('Code', () => {
  it('Should render', () => {
    expect(() => render(<Code>This is my code</Code>)).not.toThrow()
  })

  it.each([
    ['size 300', 300],
    ['size 400', 400],
    ['size 500', 500],
    ['size 600', 600]
  ])('<Code /> %s renders as expected', (_, size) => {
    const component = (
      <ThemeProvider value={defaultTheme}>
        <Code size={size}>{`Text ${size}`}</Code>
      </ThemeProvider>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it.each([
    ['size 300', 300],
    ['size 400', 400],
    ['size 500', 500],
    ['size 600', 600]
  ])('<Code /> %s with minimal appearance specified renders as expected', (_, size) => {
    const component = (
      <ThemeProvider value={defaultTheme}>
        <Code size={size} appearance="minimal">{`Text ${size}`}</Code>
      </ThemeProvider>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should pass through `className` prop', () => {
    const expected = faker.random.word().toLowerCase()
    const component = (
      <Code data-testid="code" className={expected}>
        Testing
      </Code>
    )
    const { getByTestId } = render(component)
    expect(getByTestId('code')).toHaveClass(expected)
  })
})
