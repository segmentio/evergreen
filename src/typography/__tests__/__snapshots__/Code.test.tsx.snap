// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code <Code /> size 300 renders as expected 1`] = `
Extracted Styles:
background-color: rgba(16, 112, 202, 0.06);
border-bottom-left-radius: 4px;
border-bottom-right-radius: 4px;
border-top-left-radius: 4px;
border-top-right-radius: 4px;
box-shadow: inset 0 0 0 1px rgba(16, 112, 202, 0.14);
box-sizing: border-box;
color: #474d66;
font-family: "SF Mono", "Monaco", "Inconsolata", "Fira Mono", "Droid Sans Mono", "Source Code Pro", monospace;
font-size: 12px;
font-size: 14px;
font-weight: 400;
letter-spacing: -0.05px;
letter-spacing: 0;
line-height: 16px;
line-height: 20px;
padding-bottom: 3px;
padding-left: 6px;
padding-right: 6px;
padding-top: 3px;


<code
  className="ub-fnt-sze_12px ub-f-wght_400 ub-ln-ht_16px ub-ltr-spc_0 ub-fnt-fam_tawojy ub-color_474d66 ub-bg-clr_rgba16-112-202-0-06 ub-bs_1o2cska ub-pl_6px ub-pr_6px ub-pb_3px ub-pt_3px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-box-szg_border-box"
>
  Text 300
</code>
`;

exports[`Code <Code /> size 300 with minimal appearance specified renders as expected 1`] = `
Extracted Styles:
box-sizing: border-box;
color: #474d66;
font-family: "SF Mono", "Monaco", "Inconsolata", "Fira Mono", "Droid Sans Mono", "Source Code Pro", monospace;
font-size: 12px;
font-weight: 400;
letter-spacing: 0;
line-height: 16px;


<code
  className="ub-fnt-sze_12px ub-f-wght_400 ub-ln-ht_16px ub-ltr-spc_0 ub-fnt-fam_tawojy ub-color_474d66 ub-box-szg_border-box"
>
  Text 300
</code>
`;

exports[`Code <Code /> size 400 renders as expected 1`] = `
Extracted Styles:
background-color: rgba(16, 112, 202, 0.06);
border-bottom-left-radius: 4px;
border-bottom-right-radius: 4px;
border-top-left-radius: 4px;
border-top-right-radius: 4px;
box-shadow: inset 0 0 0 1px rgba(16, 112, 202, 0.14);
box-sizing: border-box;
color: #474d66;
font-family: "SF Mono", "Monaco", "Inconsolata", "Fira Mono", "Droid Sans Mono", "Source Code Pro", monospace;
font-size: 14px;
font-weight: 400;
letter-spacing: -0.05px;
line-height: 20px;
padding-bottom: 3px;
padding-left: 6px;
padding-right: 6px;
padding-top: 3px;


<code
  className="ub-fnt-sze_14px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_tawojy ub-color_474d66 ub-bg-clr_rgba16-112-202-0-06 ub-bs_1o2cska ub-pl_6px ub-pr_6px ub-pb_3px ub-pt_3px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-box-szg_border-box"
>
  Text 400
</code>
`;

exports[`Code <Code /> size 400 with minimal appearance specified renders as expected 1`] = `
Extracted Styles:
box-sizing: border-box;
color: #474d66;
font-family: "SF Mono", "Monaco", "Inconsolata", "Fira Mono", "Droid Sans Mono", "Source Code Pro", monospace;
font-size: 14px;
font-weight: 400;
letter-spacing: -0.05px;
line-height: 20px;


<code
  className="ub-fnt-sze_14px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_tawojy ub-color_474d66 ub-box-szg_border-box"
>
  Text 400
</code>
`;

exports[`Code <Code /> size 500 renders as expected 1`] = `
Extracted Styles:
background-color: rgba(16, 112, 202, 0.06);
border-bottom-left-radius: 4px;
border-bottom-right-radius: 4px;
border-top-left-radius: 4px;
border-top-right-radius: 4px;
box-shadow: inset 0 0 0 1px rgba(16, 112, 202, 0.14);
box-sizing: border-box;
color: #474d66;
font-family: "SF Mono", "Monaco", "Inconsolata", "Fira Mono", "Droid Sans Mono", "Source Code Pro", monospace;
font-size: 16px;
font-weight: 400;
letter-spacing: -0.05px;
line-height: 20px;
padding-bottom: 3px;
padding-left: 6px;
padding-right: 6px;
padding-top: 3px;


<code
  className="ub-fnt-sze_16px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_tawojy ub-color_474d66 ub-bg-clr_rgba16-112-202-0-06 ub-bs_1o2cska ub-pl_6px ub-pr_6px ub-pb_3px ub-pt_3px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-box-szg_border-box"
>
  Text 500
</code>
`;

exports[`Code <Code /> size 500 with minimal appearance specified renders as expected 1`] = `
Extracted Styles:
box-sizing: border-box;
color: #474d66;
font-family: "SF Mono", "Monaco", "Inconsolata", "Fira Mono", "Droid Sans Mono", "Source Code Pro", monospace;
font-size: 16px;
font-weight: 400;
letter-spacing: -0.05px;
line-height: 20px;


<code
  className="ub-fnt-sze_16px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_tawojy ub-color_474d66 ub-box-szg_border-box"
>
  Text 500
</code>
`;

exports[`Code <Code /> size 600 renders as expected 1`] = `
Extracted Styles:
background-color: rgba(16, 112, 202, 0.06);
border-bottom-left-radius: 4px;
border-bottom-right-radius: 4px;
border-top-left-radius: 4px;
border-top-right-radius: 4px;
box-shadow: inset 0 0 0 1px rgba(16, 112, 202, 0.14);
box-sizing: border-box;
color: #474d66;
font-family: "SF Mono", "Monaco", "Inconsolata", "Fira Mono", "Droid Sans Mono", "Source Code Pro", monospace;
font-size: 18px;
font-weight: 400;
letter-spacing: -0.07px;
line-height: 24px;
padding-bottom: 3px;
padding-left: 6px;
padding-right: 6px;
padding-top: 3px;


<code
  className="ub-fnt-sze_18px ub-f-wght_400 ub-ln-ht_24px ub-ltr-spc_-0-07px ub-fnt-fam_tawojy ub-color_474d66 ub-bg-clr_rgba16-112-202-0-06 ub-bs_1o2cska ub-pl_6px ub-pr_6px ub-pb_3px ub-pt_3px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-box-szg_border-box"
>
  Text 600
</code>
`;

exports[`Code <Code /> size 600 with minimal appearance specified renders as expected 1`] = `
Extracted Styles:
box-sizing: border-box;
color: #474d66;
font-family: "SF Mono", "Monaco", "Inconsolata", "Fira Mono", "Droid Sans Mono", "Source Code Pro", monospace;
font-size: 18px;
font-weight: 400;
letter-spacing: -0.07px;
line-height: 24px;


<code
  className="ub-fnt-sze_18px ub-f-wght_400 ub-ln-ht_24px ub-ltr-spc_-0-07px ub-fnt-fam_tawojy ub-color_474d66 ub-box-szg_border-box"
>
  Text 600
</code>
`;
